### PR TITLE
Handle Sketch Resize Constraints (v44+)

### LINKED ISSUE #150: Add ability to define resizing attributes
Hey there! I was wondering if there was any thought around potentially adding the ability to define the resizing options on Sketch Layers. Use case would be that I'm trying to build a Symbol library that can have flexible layouts at design-time.

Screenshot below of what I'm describing:

screenshot 2017-07-12 16 36 05

### PR DESCRIPTION
**Pull request by larsonjj**
# Overview

This PR adds in full support for sketch v44+ resize constraint controls (without breaking v43 support).
screen shot 2017-09-05 at 1 26 44 am

# Changes

- Added new `resizingConstraint` prop to `View`, `Text`, `Image`, and `SymbolInstance` components
  - These props get handled by the component's respective renderers along with `SketchRenderer` base class
- Added new` ResizingConstraintPropTypes` PropType
- Added new` ResizeConstraints ` Flow type
- Updated jest snapshot tests to handle new `resizingConstraint`

# Testing this PR

Since `sketchapp-json-plugin`'s current version wont work, you'll need to setup the plugin locally. Here's how to do it:

1. git clone this repository (`react-sketchapp`), and checkout the `feature/resize_constraints` branch.
2. Install all npm dependencies: `npm install` within the react-sketchapp's root directory and then run `npm run watch`..
3. Navigate to the `react-sketchapp/examples/basic-setup`, open the `package.json` file and swap out the `"react-sketchapp"` with your local clone. It should look something like this: `"react-sketchapp": "file:{path-to-folder}/react-sketchapp",`. Then install all npm packages: `npm install`.
4. Open the `src/my-command.js` file in a text editor and add `resizingConstraint={{bottom: true, fixedWidth: true}}` property to one of the `<Text />` components
5. Open up Sketch (v44+) to a new document and run `npm run render` in the `react-sketchapp` folder within your terminal.
6. Click on the Text object within Sketch (that you added the `resizingConstraint` prop to) and notice that the resizing options panel looks like so:
screen shot 2017-09-05 at 1 26 44 am

Congrats 🎉 , you can now edit resize controls via react-sketchapp.

# Related Issues

Closes #150.

# Additional Notes

- This is a first pass API wise, by just allowing a `resizingConstraint={{ top: true, right: true, bottom: true, left: false, fixedHeight: true, fixedWidth: false }}` property on `View`, `Text`, and `Image` components. Definitely open to whatever API makes the most sense here (Props, HOC, etc)
- Prettier decided to format a couple extra files, let me know if that's an issue.

### CODE REVIEW

**Comment by jongold** [comment]
This looks awesome, thanks @larsonjj.

As for the API — do what feels right :)

@ryngonzalez you were asking for this in #150 - any opinions?

**Comment by ryngonzalez** [comment]
@jongold @larsonjj API looks good on my end! I think props makes sense for this. :shipit:

File: src/jsonUtils/hacksForJSONImpl.js [hunk 1]
```diff
@@ -95,6 +159,49 @@ export const makeImageDataFromUrl = (url: string): MSImageData => {
...
+    }
+    if (fixedWidth) {
+      constraints.push('fixedWidth');
+    }
+
+    if (constraints.length > 0) {
+      return RESIZE_CONSTRAINTS[constraints.join('_')];
+    }
+
+    return RESIZE_CONSTRAINTS.none; // No constraints
```
Comment (ryngonzalez) [inline src/jsonUtils/hacksForJSONImpl.js]: why not let this fall through to the outer `return`?

Comment (larsonjj) [reply src/jsonUtils/hacksForJSONImpl.js]: Good call! Removed.

**Comment by tbrasington** [comment]
This feature would be amazing to have! Will it make it into master soon? Is there anything I can do to help?

**Comment by larsonjj** [comment]
@jongold,

With the release of `v0.1.2` of `sketchapp-json-plugin`, this PR no longer has any extra dependencies and should be ready to go.

Let me know if there is anything else that needs changing. Thanks!

**Comment by ryngonzalez** [comment]
Wooo 🚀

**Comment by tbrasington** [comment]
Thank you! This and the yoga update are greatly appreciated 🙏

**Pull request merged**